compiler-core/src/parse.rs

```diff
@@ -2117,6 +2117,22 @@ where
         )?;
         let (_, rpar_e) =
             self.expect_one_following_series(&Token::RightParen, "a function parameter")?;
+
+        // Check for TypeScript-style return type annotation (:) instead of arrow (->)
+        if let Some((colon_start, colon_end)) = self.maybe_one(&Token::Colon) {
+            return Err(ParseError {
+                error: ParseErrorType::UnexpectedToken {
+                    token: Token::Colon,
+                    expected: vec!["->".into()],
```

Comment:
We usually wrap operators in backticks like this as it makes them easier to read

```suggestion
                    expected: vec!["`->`".into()],
```